### PR TITLE
Mention file analysis in the guides [ci skip]

### PR DESCRIPTION
**Pull request by andrewhaines**
To date, there was no mention of analyzers in the Active Storage guide. I thought saying something simple about how analysis works — especially that it's done via Active Job — was important.

As a use case, I recently spent time working to troubleshoot an application where Active Job wasn't properly configured, and therefore blob metadata wasn't populating as expected. That wasn't clear to me till I did some digging.

cc: @georgeclaghorn @peterzhu2118

### CODE REVIEW

File: guides/source/active_storage_overview.md [hunk 1]
```diff
@@ -473,6 +473,13 @@ message.video.open do |file|
 end
 ```
 
+Analyzing Files
+---------------
+
+Active Storage [analyzes](https://api.rubyonrails.org/classes/ActiveStorage/Blob/Analyzable.html#method-i-analyze) files after they've been uploaded by queuing a job in Active Job. Analyzed files will store additional information in the metadata hash, including `analyzed: true`. You can check whether a blob has been analyzed by calling `analyzed?` on it.
+
+Image analysis provides `width` and `height` attributes. Video analysis provides these, as well as `duration`, `angle`, and `display_aspect_ratio`. Video analysis also requires the [FFmpeg](https://www.ffmpeg.org/) library, which you must include separately.
```
Comment (peterzhu2118) [inline guides/source/active_storage_overview.md]: For the image analysis, the user will have to install ImageMagick and the `mini_magick` gem, you might want to mention that for the image analysis part.

**Review by peterzhu2118 — COMMENTED** [review]
Minor comment, otherwise looks solid to me!

File: guides/source/active_storage_overview.md (hunk 1 above)
Comment (kaspth) [reply guides/source/active_storage_overview.md]: That’ll be image_processing instead, the mini_magick integration is (soft-?)deprecated.

Comment (peterzhu2118) [reply guides/source/active_storage_overview.md]: Ah I see. Perhaps the documentation for `ImageAnalyzer` should be updated to reflect that? https://github.com/rails/rails/blob/master/activestorage/lib/active_storage/analyzer/image_analyzer.rb#L13

Comment (kaspth) [reply guides/source/active_storage_overview.md]: Ah yes, PR welcome! Check the Rails 6 release notes or upgrading guides. One of those should have the details.

Comment (andrewhaines) [reply guides/source/active_storage_overview.md]: @peterzhu2118 @kaspth Thanks. I've added a [new commit](https://github.com/rails/rails/pull/37751/commits/480b24264261ba18bc8af3f4215f9b61688fbb4f) to include the `image_processing` requirement for analysis. And I've slightly revised the following section on image transformation to simplify / clarify.

File: guides/source/active_storage_overview.md [hunk 2]
```diff
@@ -473,20 +473,25 @@ message.video.open do |file|
...
+---------------
 
-To create a variation of the image, call `variant` on the `Blob`. You can pass
-any transformation to the method supported by the processor. The default
-processor is [MiniMagick](https://github.com/minimagick/minimagick), but you
-can also use [Vips](https://www.rubydoc.info/gems/ruby-vips/Vips/Image).
+Active Storage [analyzes](https://api.rubyonrails.org/classes/ActiveStorage/Blob/Analyzable.html#method-i-analyze) files after they've been uploaded by queuing a job in Active Job. Analyzed files will store additional information in the metadata hash, including `analyzed: true`. You can check whether a blob has been analyzed by calling `analyzed?` on it.
 
-To enable variants, add the `image_processing` gem to your `Gemfile`:
+To enable analysis, add the `image_processing` gem to your `Gemfile`:
```
Comment (gmcgibbon) [inline guides/source/active_storage_overview.md]: This is only necessary for image analysis. The video analyzer has a dependency on `ffmpeg`. You shouldn't need any gems to enable analysis generically.

Can we create separate sections for the provided image and video analyzers?

File: guides/source/active_storage_overview.md (hunk 1 above)
Comment (georgeclaghorn) [reply guides/source/active_storage_overview.md]: `image_processing` is used for variants, not analysis. Analysis still requires `mini_magick`.

Comment (kaspth) [reply guides/source/active_storage_overview.md]: Oh ok, I totally thought we were all in on image_processing. Any particular reason we aren’t using it for analysis? Or just that it hasn’t been update for whatever reason?

Comment (georgeclaghorn) [reply guides/source/active_storage_overview.md]: `image_processing` isn’t for reading image metadata, as far as I know. It only supports transformation.

**Comment by andrewhaines** [comment]
@georgeclaghorn Thanks for the clarity. I've removed the `image_processing` reference in the analysis section and referenced `mini_magick` instead. cc: @kaspth 

@gmcgibbon I don't think separate sections for image / video analyzers is required here, unless there's more we want to say about how they work in detail. I'm happy to work on another PR with you if those sections would stretch the integrity of this PR.

**Comment by andrewhaines** [comment]
Thanks for reviewing / merging!

**Pull request merged**